framework/src/constraints/EqualValueEmbeddedConstraint.C

```diff
@@ -41,20 +44,23 @@ EqualValueEmbeddedConstraint::validParams()
   return params;
 }
 
-EqualValueEmbeddedConstraint::EqualValueEmbeddedConstraint(const InputParameters & parameters)
-  : NodeElemConstraint(parameters),
+template <bool is_ad>
+EqualValueEmbeddedConstraintTempl<is_ad>::EqualValueEmbeddedConstraintTempl(
+    const InputParameters & parameters)
+  : GenericNodeElemConstraint<is_ad>(parameters),
     _displaced_problem(parameters.get<FEProblemBase *>("_fe_problem_base")->getDisplacedProblem()),
     _fe_problem(*parameters.get<FEProblem *>("_fe_problem")),
-    _formulation(getParam<MooseEnum>("formulation").getEnum<Formulation>()),
-    _penalty(getParam<Real>("penalty")),
+    _formulation(this->template getParam<MooseEnum>("formulation").template getEnum<Formulation>()),
```

Comment:
Might have missed it, but for AD do you have an error for the kinematic formulation? Or am I wrong and the kinematic form works with AD?

Reply:
It works with AD.  i'll add the ad kinematic tests.

Reply:
I was wrong, AD with Kinematic constraints do not work.  The test is using PJFNK so it didn't need an accurate jacobian to pass.  I changed it to Newton and it fails.  I'll add an error for using kinematic with AD.